ui/src/components/LoginScreen.tsx

```diff
@@ -15,6 +15,26 @@ function useQuery() {
   return new URLSearchParams(useLocation().search);
 }
 
+function raiseParamsToHash() {
+  const url = new URL(window.location.href);
+
+  // When DABL login redirects back to app, hoist the query into the hash route.
+  // This allows react-router's HashRouter to see and parse the supplied params
+
+  // i.e., we want to turn
+  // ledgerid.projectdabl.com/?party=party&token=token/#/
+  // into
+  // ledgerid.projectdabl.com/#/?party=party&token=token
+  if (url.search !== '' && url.hash === '#/') {
+    window.location.href = `${url.origin}${url.pathname}#/${url.search}`;
+  }
+}
+
+function getTokenFromCookie(): string {
+  const tokenCookiePair = document.cookie.split('; ').find(row => row.startsWith('DABL_LEDGER_ACCESS_TOKEN')) || '';
+  return tokenCookiePair.slice(tokenCookiePair.indexOf('=') + 1);
```

Comment:
can you clarify here why we want the part after the `=` sign? the format something like `access_token=blahblah`?

Reply:
Yep, I believe that's the format of the cookie. Took the code from the docs (https://docs.projectdabl.com/api/iam/#the-login-in-with-dabl-button) that indicated the token delivery via URL query is going to eventually be phased out in favor of the cookie approach, so decided to get ahead on that in the marketplace